application/helpers/sanitize_helper.php

```diff
@@ -502,3 +502,22 @@ function sanitize_signedint($integer, $min = '', $max = '')
 
     return $int;
 };
+
+// sanitize ip address
+function sanitize_ip_address($string, $allowWildcards = true)
+{
+    if ($allowWildcards) {
+        $regex = '/^((([0-9]{1,2})|(1[0-9]{2,2})|(2[0-4][0-9])|(25[0-5])|\*)\.){3}(([0-9]{1,2})|(1[0-9]{2,2})|(2[0-4][0-9])|(25[0-5])|\*)$/';
+    } else {
+        $regex = '/^((([0-9]{1,2})|(1[0-9]{2,2})|(2[0-4][0-9])|(25[0-5]))\.){3}(([0-9]{1,2})|(1[0-9]{2,2})|(2[0-4][0-9])|(25[0-5]))$/';
+    }
+    return preg_match($regex, $string, $m) ? $string : '';
+}
+
+function check_ip_address($input, $allowWildcards = true)
```

Comment:
Missing docblock :( And above

Reply:
To be a little anal, can you maybe do this as well?

> The first line in the docblock should be one sentence describing the relation between function arguments and its output

https://manual.limesurvey.org/Code_quality_guide#Contract

Reply:
> Checks the validity of an IPv4 address 

would be

> Checks the validity of an IPv4 address $input

Or maybe, with renaming the argument:

> Checks the validity of IPV4 address $ip

Reply:
Boolean arguments that changes the "mode" of the function can be hard to fit in a one-liner like that. :)

Reply:
> Boolean arguments that changes the "mode" of the function can be hard to fit in a one-liner like that

What do you mean? Sorry! Description for that argument would be too long?